pkg/controller/admin/superusers.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/exposure-notifications-verification-server/internal/auth"
+	"github.com/google/exposure-notifications-verification-server/internal/project"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/gorilla/mux"
+)
+
+// HandleSuperUsersIndex renders the list of system admins.
+func (c *Controller) HandleSuperUsersIndex() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		admins, err := c.db.ListSystemAdmins()
+		if err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		m := controller.TemplateMapFromContext(ctx)
+		m["admins"] = admins
+		c.h.RenderHTML(w, "admin/superusers/index", m)
+	})
+}
+
+// HandleSuperUsersCreate creates a new system admin.
+func (c *Controller) HandleSuperUsersCreate() http.Handler {
+	type FormData struct {
+		Email string `form:"email"`
+		Name  string `form:"name"`
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		session := controller.SessionFromContext(ctx)
+		if session == nil {
+			controller.MissingSession(w, r, c.h)
+			return
+		}
+		flash := controller.Flash(session)
+
+		currentUser := controller.UserFromContext(ctx)
+		if currentUser == nil {
+			controller.MissingUser(w, r, c.h)
+			return
+		}
+
+		// Requested form, stop processing.
+		if r.Method == http.MethodGet {
+			var user database.User
+			c.renderNewUser(ctx, w, &user)
+			return
+		}
+
+		var form FormData
+		err := controller.BindForm(w, r, &form)
+		email := project.TrimSpace(form.Email)
```

Comment:
This trimming and validation should happen in the model via a `BeforeSave`, not the controller.

Reply:
That does also happen in BeforeSave, but FindUserByEmail needs the trim

Reply:
Why not trim in `FindUserByEmail` then?